### PR TITLE
Reference data by FAC=1.1.4

### PR DESCRIPTION
**Pull request by fujiisoup**
@mfgu 

Based on [this comment](https://github.com/flexible-atomic-code/fac/pull/4#issuecomment-376542807) in flexible-atomic-code/fac#4, 
I would like to store the reference data in this repository and keep checking the code health of fac by the automatic testing.

I think the good reference data are on https://www-amdis.iaea.org/FAC/,
Do you have an input file used to create these data?

### CODE REVIEW

File: scripts/make_data.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            os.mkdir(dir1)
+
+        (temp, logt, population) = get_tgrid(z, nele, dt=1)
+        print('NION = 1')
+        spectrum([nele], temp, den, population,
+                 asym, dir0=dir0, dir1=dir1, nion=1)
+
+
+make_spectrum('He', [1, 2], 2)
+make_spectrum('Li', [1, 2, 3], 3)
```
Comment (fujiisoup) [inline scripts/make_data.py:33]: I added a simple script similar to that in `demo` directory, but it fails for neutral Lithium.

**Review by fujiisoup — COMMENTED** [review]
The bunch of files in this PR is calculated with FAC=1.1.4 (the latest release version), but the result may depend on the input parameters.

I think it would be nice if we keep them similar to those in the FAC cite.
In the long term, the fac website may cite these data in this repo directly.

**Comment by fnevgeny** [comment]
I don't think this is a good idea. These data are complex/large. You want to start with simple calculations. Again, the "demo" directory might be a good start.

**Comment by fujiisoup** [comment]
I considered simple elements in that website, such as helium or lithium,...
but as you suggest, it would be a better idea to use demos. (I just noticed that pre-computed results are included in this repository.)

I think I tried to do too many things at once. 
OK. Closing this PR.

**Comment by fnevgeny** [comment]
> (I just noticed that pre-computed results are included in this repository.)

These need to be updated, since they were calculated with previous (some very old) versions of FAC, and many numerics have changed since.

You can backport the "make check" stuff rather straightforwardly from cfac. Of course, the "reference" data will be slightly different & need to be calculated with 1.1.4 - again, because of the different numerics.

**Comment by fujiisoup** [comment]
I looked inside `make check` stuff in cfac. It looks that you compared texts in two files. 
I'm curious how these tests are robust against the code update?
I thought that updating a logic would introduce a negligible but finite difference in the result.

**Comment by fnevgeny** [comment]
> It looks that you compared texts in two files.

Correct. This is certainly sub-optimal.

> I'm curious how these tests are robust against the code update?

So far it has been OK, also across compiler versions and OS's (Linux/Windows/Mac).

> I thought that updating a logic would introduce a negligible but finite difference in the result.

What do you mean by "updating a logic"?

**Comment by fujiisoup** [comment]
> What do you mean by "updating a logic"?

Any numerical thing.
e.g. 
If we have a loop to compute a summation of an array and if we update the computation order for speed up calculation (e.g. reversed order),
the amplitude of the round-off error may change, though its amount might be negligible.

I guess you made such optimizations for cfac and then the result varied.
Did you update the reference files every time?

I think backporting your make-check machinary is the first step, but I am wondering if we may need to keep the reference files as long as possible not to confuse such numerical issues against mis-computation.

**Comment by fnevgeny** [comment]
Compiler optimizations should not affect the results; we compare results printed with much lower than hardware FP accuracy.

> I guess you made such optimizations for cfac and then the result varied.

No, I replaced all Fortran numerics with GSL - some also using different algorithms.

> Did you update the reference files every time?

Only when results differed :)

**Pull request closed without merging**